app/src/main/java/com/frinder/frinder/utils/LocationUtils.java

```diff
@@ -0,0 +1,116 @@
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.SettingsClient;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+
+import static com.google.android.gms.location.LocationServices.getFusedLocationProviderClient;
+
+public class LocationUtils {
+    private Context context;
+    private LocationRequest mLocationRequest;
+    private static final String TAG = "LocationUtil";
+    private static final int REQUEST_FINE_LOCATION = 99;
+    private long UPDATE_INTERVAL = 10 * 60 * 1000;  /* 10 mins */
+    private long FASTEST_INTERVAL = 60*1000; /* 60 sec */
+
+
+    public LocationUtils(Context context) {
+        this.context = context;
+    }
+
+    public interface LocationUpdate {
+        void onSuccess(Location location);
+    }
+
+    public void getLastLocation(final LocationUpdate callback) {
+        if (ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions();
+            return;
+        } else {
+            requestLocation(callback);
+        }
+    }
+
+    private void requestLocation(final LocationUpdate callback) {
+        FusedLocationProviderClient locationClient = getFusedLocationProviderClient(context);
+        locationClient.getLastLocation()
+                .addOnSuccessListener(new OnSuccessListener<Location>() {
+                    @Override
+                    public void onSuccess(Location location) {
+                        // GPS location can be null if GPS is switched off
+                        if (location != null) {
+                            callback.onSuccess(location);
+                        } else {
+                            Toast.makeText(context, "Please enable GPS!", Toast.LENGTH_LONG).show();
+                        }
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.d(TAG, "Error trying to get last GPS location");
+                        e.printStackTrace();
+                    }
+                });
+    }
+
+    // Trigger new location updates at interval
+    public void startLocationUpdates() {
```

Comment:
We may be responsible for stopping the location updates as well. Is that not the case?

Reply:
yes..got to add that..